hrabsence/CRM/HRAbsence/BAO/HRAbsenceType.php

```diff
@@ -169,10 +189,49 @@ public static function del($absenceTypeId) {
       CRM_Core_BAO_OptionValue::del($id);
     }
 
+    // Delete Absence Type
     $absenceType->delete();
 
     CRM_Utils_Hook::post('delete', 'HRAbsenceType', $absenceTypeId, $absenceType);
   }
+  
+  /**
+   * Count leave requests for given activity type id
+   * 
+   * @param int $activityTypeID
+   *   ID of activity type to be used to count
+   * @return int
+   *   Number of activities found with given activity type ID
+   */
+  private static function countLeaveRequests($activityTypeID) {
+    return civicrm_api3('Activity', 'getcount', [
+      'sequential' => 1,
+      'activity_type_id' => $activityTypeID
+    ]);
+  }
+
+  /**
+   * Deletes all leave entitlements for given absence type ID.
+   * 
+   * @param int $absenceTypeId
+   *   ID of Absence Type for which entitlements have to be deleted
+   */
+  private static function deleteLeaveEntitlements($absenceTypeId) {
+    $result = civicrm_api3('HRJobLeave', 'get', [
+      'debug' => 1,
+      'sequential' => 1,
+      'leave_type' => $absenceTypeId,
+      'jobcontract_revision_id' => array('>' => 0)
+    ]);
+    foreach ($result['values'] as $currentLeave) {
+      $result = civicrm_api3('HRJobLeave', 'delete', [
+        'debug' => 1,
```

Comment:
Why debug? Also, I don't think sequential will make any difference for a delete operation